parcels/field.py

```diff
@@ -490,15 +501,26 @@ def cell_areas(self):
     def search_indices_vertical_z(self, z):
         grid = self.grid
         z = np.float32(z)
-        if z < grid.depth[0]:
-            raise FieldOutOfBoundSurfaceError(0, 0, z, field=self)
-        elif z > grid.depth[-1]:
-            raise FieldOutOfBoundError(0, 0, z, field=self)
-        depth_index = grid.depth <= z
-        if z >= grid.depth[-1]:
-            zi = len(grid.depth) - 2
+        if grid.depth[-1] > grid.depth[0]:
+            if z < grid.depth[0]:
+                raise FieldOutOfBoundSurfaceError(0, 0, z, field=self)
+            elif z > grid.depth[-1]:
+                raise FieldOutOfBoundError(0, 0, z, field=self)
+            depth_index = grid.depth <= z
```

Comment:
minor formulation feedback: this function does not return just 1 index, but a mask field for depth indices, so it may be (for comprehension) better to call the variable `depth_indices`. Just a suggestion.